2_process/src/match_sites_reaches.R

```diff
@@ -0,0 +1,48 @@
+#' original author: David Watkins
+#' from https://code.usgs.gov/wwatkins/national-site-reach/-/blob/master/R/match_sites_reaches.R
+#' modified by: Jeff Sadler
+#' Match each site with a reach (seg_id/COMID)
+get_site_flowlines <- function(reach_sf, sites) {
+  #' 
+  #' @description Function to match reaches (flowlines) to point sites 
+  #'
+  #' @param reach_sf sf object of reach polylines with column "segidnat" and in WGS84
+  #' @param sites dataframe with columns "lat" "lon"
+  #'
+  #' @value A data frame the same columns as the sites input dataframe with additional columns
+  #' of "segidnat" and "offset" where "offset" is the distance (in degrees) between the point
+  #' and matching flowline
+  
+  reaches_nhd_fields <- reach_sf %>%
+    select(COMID = segidnat) %>%
+    mutate(REACHCODE = COMID, ToMeas = 100, FromMeas = 100) %>%
+    st_as_sf()
+  
+  sites_sf <- sites %>% rowwise() %>%
+    filter(across(c(lon, lat), ~ !is.na(.x))) %>%
+    mutate(Shape = list(st_point(c(lon, lat), dim = "XY"))) %>%
+    st_as_sf() %>% st_set_crs(4326) %>%
```

Comment:
Our sitelist is referenced to NAD83 rather than WGS84 as is coded here. I suggest we either add an argument to the function that prompts the user to enter the input datum or parse that information from the site table within the function. We do something similar to the latter option in the `transform_site_locations()` function.

Reply:
good idea